api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/hiv/art/InitialArtStartDateCalculation.java

```diff
@@ -35,18 +41,25 @@
 	public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> parameterValues,
 	                                     PatientCalculationContext context) {
 
-		// Get earliest dates from orders
-		Concept arvs = Dictionary.getConcept(Dictionary.ANTIRETROVIRAL_DRUGS);
-		CalculationResultMap allDrugOrders = allDrugOrders(arvs, cohort, context);
-		CalculationResultMap earliestOrderDates = earliestStartDates(allDrugOrders, context);
-
-		// Return the earliest of the two
-		CalculationResultMap result = new CalculationResultMap();
+		Date startDate = null;
+		CalculationResultMap ret = new CalculationResultMap();
 		for (Integer ptId : cohort) {
-			Date orderDate = EmrCalculationUtils.datetimeResultForPatient(earliestOrderDates, ptId);
+			Encounter lastDrugRegimenEditorEncounter = EncounterBasedRegimenUtils.getLastEncounterForCategory(Context.getPatientService().getPatient(ptId), "ARV");   //last DRUG_REGIMEN_EDITOR encounter
```

Comment:
@patryllus please change this to fetch the first encounter. Your approach will fetch the last one

Reply:
Thanks noted. Updating in a few.